apps/api/src/domain/models/song/commands/add-lyrics-for-song/add-lyrics-for-song.command.integration.spec.ts

```diff
@@ -0,0 +1,94 @@
+import getValidAggregateInstanceForTest from '../../../../../domain/__tests__/utilities/getValidAggregateInstanceForTest';
+import { IIdManager } from '../../../../../domain/interfaces/id-manager.interface';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { NotFound } from '../../../../../lib/types/not-found';
+import TestRepositoryProvider from '../../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import generateDatabaseNameForTestSuite from '../../../../../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import { DTO } from '../../../../../types/DTO';
+import { assertCommandSuccess } from '../../../__tests__/command-helpers/assert-command-success';
+import { CommandAssertionDependencies } from '../../../__tests__/command-helpers/types/CommandAssertionDependencies';
+import { dummySystemUserId } from '../../../__tests__/utilities/dummySystemUserId';
+
+const commandType = 'ADD_LYRICS_FOR_SONG';
+
+const existingSong = getValidAggregateInstanceForTest(AggregateType.song).clone({
+    lyrics: undefined,
+});
+
+const validPayload: AddLyricsForSong = {
+    aggregateCompositeIdentifier: existingSong.getCompositeIdentifier(),
+    lyrics: existingSong.lyrics,
+    languageCode: LanguageCode,
+};
+
+const validCommandFSA = {
+    type: commandType,
+    payload: validPayload,
+};
+
+const buildValidCommandFSA = (): FluxStandardAction<DTO<AddLyricsForSong>> => validCommandFSA;
+
+describe(commandType, () => {
+    let app: INestApplication;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeAll(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    describe('When the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA,
+                initialState: new DeluxeInMemoryStore({
+                    [AggregateType.song]: [existingSong],
+                }).fetchFullSnapshotInLegacyFormat(),
+                checkStateOnSuccess: async ({
+                    aggregateCompositeIdentifier: { id: SongCompositeId },
```

Comment:
```ts
{id: songId}
```

Note two things here:
1. We only capitalize names of types \ classes enums, and `id` is a (string) value
2. `id` is a plain (string uuid) ID, not a composite ID. Recall that (in our system) a composite ID is an object of the form:
```ts
{
    type: string;
    id: string;
}
```

an example would be
```ts
{
   type: 'song', // or AggregateType.song which is just a fancy reference to the string 'song'
   id: '17f66d21-df73-4544-9100-6a67f4888809'
}
```